pkg/generators/repo_host_test.go

```diff
@@ -0,0 +1,74 @@
+package generators
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	argoprojiov1alpha1 "github.com/argoproj-labs/applicationset/api/v1alpha1"
+	"github.com/argoproj-labs/applicationset/pkg/services/repo_host"
+)
+
+func TestRepoHostGetSecretRef(t *testing.T) {
```

Comment:
It would be nice to reafctor this to [table tests](https://github.com/golang/go/wiki/TableDrivenTests)
and to add a name for each case

Reply:
Updated :)